core/src/test/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriterSuite.java

```diff
@@ -562,4 +562,18 @@ public void testPeakMemoryUsed() throws Exception {
     }
   }
 
+  @Test
+  public void writeEmptyIteratorNotCreateEmptySpillFile() throws Exception {
```

Comment:
This looks like mostly a duplication of `writeEmptyIterator`. Logically, do we need to keep both `writeEmptyIterator` and `writeEmptyIteratorNotCreateEmptySpillFile` seperately?